curator-x-async/src/main/java/org/apache/curator/x/async/modeled/details/ModeledCacheImpl.java

```diff
@@ -148,38 +165,32 @@ public void childEvent(CuratorFramework client, TreeCacheEvent event)
         }
     }
 
-    private void internalChildEvent(TreeCacheEvent event) throws Exception
+    private void internalChildEvent(TreeCacheEvent event)
```

Comment:
In case of `TreeCache`, this code can called from `Executor`, but in case of `CuratorCache`, it can be called only on the event thread. No way it can be an issue? 
It could add itself as listener also using `addListener(T listener, Executor executor)`

Reply:
> but in case of CuratorCache, it can be called only on the event thread

No, it isn't. Listeners are called via `client.runSafe(() -> listenerManager.forEach(proc));`